manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -542,23 +546,26 @@ func (r *FybrikApplicationReconciler) checkGovernanceActions(configEvaluatorInpu
 	var err error
 	switch configEvaluatorInput.Request.Usage {
 	case taxonomy.WriteFlow:
-		if !req.Context.Requirements.FlowParams.IsNewDataSet {
+		if !req.Context.Requirements.FlowParams.IsNewDataSet ||
+			req.Context.Requirements.FlowParams.ResourceMetadata != nil {
 			// update an existing dataset
 			// query the policy manager whether the operation is allowed
 			reqAction := policymanager.RequestAction{
 				ActionType:         configEvaluatorInput.Request.Usage,
 				Destination:        req.DataDetails.ResourceMetadata.Geography,
```

Comment:
req.DataDetails does not have Geography for a new asset.

Reply:
ok, it set to empty string in this case...